tf/main.tf

```diff
@@ -12,169 +12,123 @@
 //  See the License for the specific language governing permissions and
 //  limitations under the License.
 
-provider "google" {
-  project = "PROJECT_ID"
-}
+terraform {
+  required_version = ">= 0.13"
 
-resource "google_project_service" "artifactregistry_googleapis_com" {
-  service = "artifactregistry.googleapis.com"
+  required_providers {
+    google = {
+      version = "~> 4.36"
+      source  = "hashicorp/google"
+    }
+  }
 }
 
-resource "google_project_service" "binaryauthorization_googleapis_com" {
-  service = "binaryauthorization.googleapis.com"
-}
+resource "random_id" "suffix" {
 
-resource "google_project_service" "cloudbuild_googleapis_com" {
-  service = "cloudbuild.googleapis.com"
+  prefix = var.project_name
+  byte_length = 4
 }
 
-resource "google_project_service" "clouddeploy_googleapis_com" {
-  service = "clouddeploy.googleapis.com"
-}
+resource "google_project" "generated_project" {
+  count           = var.existing_project_id == null ? 1 : 0
+  project_id      = local.gen_project_id
+  name            = local.gen_project_id
+  billing_account = var.google_billing_account
 
-resource "google_project_service" "container_googleapis_com" {
-  service = "container.googleapis.com"
+  lifecycle {
+    # ignoring org_id changes allows the project to be created in whatever org
+    # the user is part of by default, without having to explicitly include the
+    # org id in the terraform config.
+    ignore_changes = [org_id]
+    precondition {
+      condition     = var.google_billing_account != null
+      error_message = "If an existing Project ID is not specified, you must provide a billing account that we can use to create one for you"
+    }
+  }
 }
 
-resource "google_project_service" "containeranalysis_googleapis_com" {
-  service = "containeranalysis.googleapis.com"
+
+provider "google" {
+  project = local.final_project_id
 }
-resource "google_project_service" "containerscanning_googleapis_com" {
-  service = "containerscanning.googleapis.com"
+
+data "google_project" "project_info" {
+  project_id = local.final_project_id    
 }
 
-resource "google_project_service" "containersecurity_googleapis_com" {
-  service = "containersecurity.googleapis.com"
+## Enable Services
+
+resource "google_project_service" "sds_demo_services" {
+  project = local.final_project_id
+  service = "${each.value}.googleapis.com"
+
+  for_each = toset([
+    "artifactregistry",
+    "binaryauthorization",
+    "cloudbuild",
+    "clouddeploy",
+    "container",
+    "containeranalysis",
+    "containerscanning",
+    "containersecurity",
+    "run"
+  ])
 }
 
-resource "google_project_service" "run_googleapis_com" {
-  service = "run.googleapis.com"
+resource "time_sleep" "wait_for_services" {
+  create_duration = "30s"
```

Comment:
Increase maybe to 60 or 120s?

Reply:
The Cloud Deploy API doesn't seem to return that it's enabled. Did you experience this?

Reply:
Yeah, in some cases it can take a while. I'll bump this up.